### PR TITLE
5356. add showAPIShare prop to Maps plugin, upd withSharetool

### PR DESCRIPTION
**Pull request by gitcontributor**
## Description

**Please check if the PR fulfills these requirements**
- [ x ] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ x ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue
**What is the current behavior?**

#5356  

**What is the new behavior?**

Added showAPIShare property and updated withShareTool to be able to handle the shareAPI visibility from local config in maps plugin.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Comment by gitcontributor** [comment]
This PR was created to fix this issue: https://github.com/geosolutions-it/MapStore2-C028/issues/127
The problem is that the Maps plugin is used on the home page in the map contents to handle share component instead of Share plugin, so i had to add a new property so that i could configure visibility of share api from localConfig->maps plugin.

File: web/client/components/resources/enhancers/withShareTool.jsx [hunk 1]
```diff
@@ -11,20 +11,26 @@ import {compose, withState} from 'recompose';
...
     const shareUrlResult = editedResource ? getShareUrl(editedResource) : '';
     const resourceUrl = isString(shareUrlResult) ? shareUrlResult : shareUrlResult.url;
-    const showAPI = isString(shareUrlResult) ? shareApi : shareUrlResult.shareApi;
     // window.location
     const location = getLocationObject();
     const baseURL = location && (location.origin + location.pathname);
     const fullUrl = editedResource ? baseURL + '#/' + resourceUrl : '';
 
+    let showAPI;
+    if (!isNil(showAPIShare)) {
```
Comment (MV88) [inline web/client/components/resources/enhancers/withShareTool.jsx:28]: Please document in this enhancer the showAPIShare property

i mean inside "Adds sharing functionality to a resource grid."

File: web/client/components/resources/enhancers/withShareTool.jsx [hunk 2]
```diff
@@ -54,6 +60,7 @@ export const addSharePanel = Component => props => {
 * @prop {boolean} shareApi: controls, whether Share Dialog should include an option if embedding with APIs
 * @prop {object} [shareOptions] options to pass to the SharePanel
 * @prop {function} [getLocationObject] method to retrieve window.location. If not passed, window.location will be used.  (Overridable by unit tests)
+* @prop {boolean} showAPIShare: controls, whether Share Dialog should show API in the Maps grid
```
Comment (MV88) [inline web/client/components/resources/enhancers/withShareTool.jsx:63]:
```suggestion
* @prop {boolean} showAPIShare: controls, whether Share Dialog should show API (it wins over the other flags)
```

File: web/client/components/resources/enhancers/withShareTool.jsx [hunk 3]
```diff
@@ -14,16 +14,16 @@ import ShareUtils from '../../../utils/ShareUtils';
 import {isString} from 'lodash';
 
 export const addSharePanel = Component => props => {
-    const { showShareModal, onShowShareModal, shareModalSettings, setShareModalSettings, editedResource, setEditedResource, shareOptions = {}, getLocationObject = () => window.location, ...other } = props;
+    const { showAPIShare, showShareModal, onShowShareModal, shareModalSettings, setShareModalSettings, editedResource, setEditedResource, shareOptions = {}, getLocationObject = () => window.location, ...other } = props;
```
Comment (MV88) [inline web/client/components/resources/enhancers/withShareTool.jsx:17]:
```suggestion
    const { showShareModal, onShowShareModal, shareModalSettings, setShareModalSettings, editedResource, setEditedResource, shareOptions = {}, getLocationObject = () => window.location, ...other } = props;
```

**Comment by MV88** [comment]
@gitcontributor please backport this pr

**Pull request merged**